src/legendre_polynomial.cpp

```diff
@@ -0,0 +1,83 @@
+#include <tallies/legendre_polynomial.hpp>
+
+#include <iostream>
+
+LegendrePolynomials::LegendrePolynomials(std::size_t order)
+    : legendre_coeff_(), order_(order) {
+  // evaluate the coefficient for each and every order
+  // from zero to given order
+
+  // get the total number of coefficient to store
+  std::size_t coeff_length = 1;
+  if (order_ != 0) {
+    if (order_ % 2 == 0) {
+      const std::size_t length =
+          static_cast<std::size_t>((static_cast<double>(order_) - 1.) * 0.5 + 1.);
+      coeff_length =
+          length * (length + 1) + static_cast<std::size_t>(static_cast<double>(order_) * 0.5 + 1);
+    } else {
+      const std::size_t length = static_cast<std::size_t>(static_cast<double>(order_) * 0.5 + 1);
```

Comment:
Here, order_ isn't even, and we are still multiplying by 0.5. Are we sure this cast from double to std::size_t is safe ?